exec/java-exec/pom.xml

```diff
@@ -609,6 +559,59 @@
           </execution>
         </executions>
       </plugin>
+      <plugin> <!-- source file must end up in the jar for janino parsing -->
+        <groupId>org.codehaus.mojo</groupId>
+        <artifactId>build-helper-maven-plugin</artifactId>
+        <version>1.9.1</version>
+        <executions>
+          <execution>
+            <id>add-sources-as-resources</id>
+            <phase>process-sources</phase>
+            <goals>
+              <goal>add-resource</goal>
+            </goals>
+            <configuration>
+              <resources>
+                <resource>
+                  <directory>src/main/java</directory>
+                </resource>
+                <resource>
+                  <directory>src/test/java</directory>
```

Comment:
I don't think we need to the test directory. Definitely not for the distribution jars.